File: csharp-ds/utils/TestRunner.cs
using System;

public static class TestRunner
{
    private static int _passed;
    private static int _failed;

    public static void Run(string name, Action test)
    {
        try
        {
            test();
            _passed++;
            Console.WriteLine($"PASS: {name}");
        }
        catch (Exception ex)
        {
            _failed++;
            Console.WriteLine($"FAIL: {name} -> {ex.Message}");
        }
    }

    public static void Summary(string suite)
    {
        Console.WriteLine($"Suite: {suite}");
        Console.WriteLine($"Passed: {_passed}");
        Console.WriteLine($"Failed: {_failed}");
    }
}

File: typescript-ds/utils/test-runner.ts
type TestCase = {
  name: string;
  run: () => void;
};

let passed = 0;
let failed = 0;

export function runTest(testCase: TestCase): void {
  try {
    testCase.run();
    passed++;
    console.log(`PASS: ${testCase.name}`);
  } catch (error) {
    failed++;
    const message = error instanceof Error ? error.message : String(error);
    console.error(`FAIL: ${testCase.name} -> ${message}`);
  }
}

export function printSummary(suiteName: string): void {
  console.log(`\nSuite: ${suiteName}`);
  console.log(`Passed: ${passed}`);
  console.log(`Failed: ${failed}`);

  if (failed > 0) {
    process.exitCode = 1;
  }
}

export function resetSummary(): void {
  passed = 0;
  failed = 0;
}
